docs/RELEASE.md

```diff
@@ -0,0 +1,62 @@
+# Processes
+
+The following processes are necessary to actualize our releases. Each process has a *Cadence* on which it must execute and an *Responsible* that is responsible for autonomously doing so and reporting back any error in the RelEng<sup>1</sup> channel.
+
+## Mainline Release
+
+Cadence: every two weeks. Responsible: Release Team.
+
+This process aims to release the `stable` branch as a *Mainline* release every two weeks. It needs to start three days before the proposed release deadline to allow leeway for unexpected issues. This process should eventually be automated.
+
+### Steps
+
+1. [ ] Check if process [Clobbering](#clobbering) needs to happen and do so first, if that is the case.
+1. [ ] Check out the latest commit of `stable`.
+2. [ ] Verify all CI checks of that commit.
+3. [ ] Announce that commit as cutoff *Mainline* for a release in the General<sup>2</sup> chat.
+4. [ ] Bump the semver of all crates <!-- FAIL-CI: We need some better process here on how to do it exactly -->
```

Comment:
This should be really not done. IMO we should bump the crate versions as we merge stuff to `unstable`.

I had written here some things: https://github.com/paritytech/polkadot-sdk/pull/2366

I mean I know that is not a perfect thing, but probably the best we can do. IMO this should not be done by a single person or after the fact as this is otherwise too hard to follow.

Reply:
But if we bump the version of each crate (and all its transitive dependants) in each MR, then we will have huge merge conflicts.  
I think if we instead note the version bump in the `prdoc` then it can happen automatically at release process.

Reply:
> It would also require is to backport the version bumps from crates-io, leading to even more conflicts.

Hmm? Why would we need to backport version bumps? I mean the version isn't bumping automatically. 



> I think if we instead note the version bump in the `prdoc` then it can happen automatically at release process.

If you change something deep inside the tree, I don't want to copy past 300 crate names and their bumps to a prdoc file...



> But if we bump the version of each crate (and all its transitive dependants) in each MR, then we will have huge merge conflicts.

We should not bump in every pr. Bumps should only be done as they are required between releases.

Reply:
>Hmm? Why would we need to backport version bumps? I mean the version isn't bumping automatically.

Yea I see. I thought about using `prdoc` to not the semver bump of each crate that i changed, and a script can then take the maximum of all bumps and apply them.

>If you change something deep inside the tree, I don't want to copy past 300 crate names and their bumps to a prdoc file...

It could suffice to note the changed crates - all dependant ones could bump automatically.

>We should not bump in every pr. Bumps should only be done as they are required between releases.

Okay yea i read it in the other MR from you. So:
- developer changes one crate and bumps that.  
- CI reminds him to not bump multiple times but to also bump all dependant crates
- CI checks that there are no obvious and unintended breaking changes

Reply:
> It could suffice to note the changed crates - all dependant ones could bump automatically.

Bumping the dependent crates is not that easy. I mean you only need to bump the dependent crates if you re-export anything as public api.

Reply:
But when we change something in `sp-io` for example, without modifying the public API but some implementation detail, then we should bump all dependants, or?  
I am not sure if its easy to manually select which crates to bump, since there is the chance of missing some. If we always bump all dependants then maybe we bump too many, but at least we never miss any.

Reply:
I changed it now to make the developer update the crates versions in their Merge Requests (where necessary).

Reply:
FYI working on a tool for this https://github.com/liamaharon/cargo-workspace-version-tools

Reply:
> Bumping the dependent crates is not that easy. I mean you only need to bump the dependent crates if you re-export anything as public api.

This isn't true when making breaking version bumps. Types are not the same between major versions, even if they have the exactly same method signature and api https://doc.rust-lang.org/cargo/reference/resolver.html#version-incompatibility-hazards

I just stumbled across this https://github.com/dtolnay/semver-trick/tree/master which 

> refers to publishing a breaking change to a Rust library without requiring a coordinated upgrade across its downstream dependency graph.

Kinda hacky, but what we need to not be stuck in the hell of major bumping dependant crates.. Would need to some time spent thinking about how to implement this with minimum friction, however.